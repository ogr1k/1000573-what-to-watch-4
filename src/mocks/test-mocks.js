import {getAvailableGenres} from "../utils.js";

const FILMS_MOCKS_QUANTITY = 8;

export const PROMOFILM = {
  name: `BATMAN`,
  genre: `ACTION`,
  date: 2014,
  image: `https://i1.wp.com/eisenhowerlibrary.org/wp-content/uploads/2018/04/Batman-Gotham-by-Gaslight-2018-movie-poster1-220x330.jpg`
};

export const film = {
  name: `Jojo Rabbit`,
  poster: ``,
  description: ``,
  director: ``,
  genre: ``,
  rating: 1,
  ratings: 1,
  starring: ``,
  year: 1,
  video: ``,
};

export const films = [];

for (let i = 0; i < FILMS_MOCKS_QUANTITY; i++) {
  const copiedFilm = Object.assign({}, film);
  copiedFilm.id = i;

  films[i] = copiedFilm;
}

export const availableGenres = getAvailableGenres(films);
