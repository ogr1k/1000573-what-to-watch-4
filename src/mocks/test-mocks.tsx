import {getAvailableGenres} from "../utils.js";
import {Film} from "../types.js";

const FILMS_MOCKS_QUANTITY = 8;

export const film: (Film) = {
  description: `A former Prohibition-era Jewish gangster returns to the Lower East Side of Manhattan over thirty years later, where he once again must confront the ghosts and regrets of his old life.`,
  director: `Sergio Leone`,
  genre: `Crime`,
  id: 1,
  name: `Once Upon a Time in America`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg`,
  rating: 10,
  year: 1984,
  ratings: 276395,
  backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg`,
  backgroundColor: `#fff`,
  starring: [`Robert De Niro`, `James Woods`, `Elizabeth McGovern`],
  runTime: 500,
  mainVideo: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  isFavourite: false,
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
};

export const films = [];

for (let i = 0; i < FILMS_MOCKS_QUANTITY; i++) {
  const copiedFilm = Object.assign({}, film);
  copiedFilm.id = i;

  films[i] = copiedFilm;
}

export const availableGenres = getAvailableGenres(films);
