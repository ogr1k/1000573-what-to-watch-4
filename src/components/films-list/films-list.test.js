import React from "react";
import renderer from "react-test-renderer";
import FilmsList from "./films-list.jsx";
import {FILMS} from "../../mocks/test-mocks.js";


const cardClickHandler = () => {};

it(`Render Films-List`, () => {
  const tree = renderer
    .create(< FilmsList films={FILMS} onClick={cardClickHandler} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
