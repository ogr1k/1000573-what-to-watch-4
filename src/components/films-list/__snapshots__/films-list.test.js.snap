// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Films-List 1`] = `
<div
  className="catalog__movies-list"
/>
`;
